lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1926,8 +1927,21 @@ internal class EvaluatingCompiler(
         }
     }
 
-    private fun compileExec(node: ExprNode): ThunkEnv {
-        TODO()
+    private fun compileExec(node: Exec): ThunkEnv {
+        val (procedureName, args, metas: MetaContainer) = node
+        val argThunks = args.map { compileExprNode(it) }
+        val procedure = procedures[procedureName.name] ?: err(
+            "No such stored procedure: ${procedureName.name}",
+            ErrorCode.EVALUATOR_NO_SUCH_PROCEDURE,
+            errorContextFrom(metas).also {
+                it[Property.PROCEDURE_NAME] = procedureName.name
+            },
+            internal = false)
+
+        return thunkFactory.thunkEnv(metas) { env ->
+            val procedureArgValues = argThunks.map { it(env) }
+            procedure.call(env, procedureArgValues)
```

Comment:
Validate the arity & throw here!

Reply:
I put the arity validation and throwing after checking if the given stored procedure exists ([L1941](https://github.com/partiql/partiql-lang-kotlin/blob/d618ba5239f05d59112679400b92f2f1b8ba19bc/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L1941)). Would it be better to put it in the thunk?